Master

nbhaan opened the pull request:
First prototype

nbhaan committed:
css files and index

pictures of the day cycle and updated index.html

nbhaan committed:
implementing buttons reveal and next

showing answer on reveal, going to next word on next

nbhaan committed:
First prototype

nbhaan committed:
log length of word

being able to print the length of a word in pixels

Nienhuis committed:
extra text

nbhaan committed:
Reveal button

Multiple clicking reveal resulted in bold font

mircealungu commented on js/main.js:
```diff
@@ -47,41 +57,99 @@ function startNewSession() {
 }
 
+function printDigitalTime() {
+	'use strict';
```
please verify if it's sufficient to use 'use strict' once per file.

mircealungu commented on js/main.js:
```diff
@@ -70,14 +70,18 @@ window.onload = function() {
     	ctx.clearRect(0,0, 380,380);
     	printOnScreen(words[wordNumber], 180, 50);
-        printOnScreen(translations[wordNumber], 180, 100);
+        
```
no magic numbers in the code. 
extract these into constants.

Nienhuis committed:
refactoring constants and strings

nbhaan committed:
revealClicked deleted

nbhaan committed:
Small changes

SCREEN_MIDDLE, adding icon.png, changing name of the app

nbhaan committed:
Word length, refactoring, layout

Words or sentences that are too long won’t be shown anymore.
CanvasWords, Time, ctxWords, ctxTime multiple time initialized, now
fixed. Different layout, bigger time, bigger words

Nienhuis committed:
Added login screen

User is able to fill in 4 digits before entering the app. After the next
button is clicked, the words will be shown as usual.

nbhaan committed:
New test layout

Trying to get bigger time

Nienhuis committed:
background changed for new ratios

clock space is changed: 100-160px
button is changed: 120 - 60px

nbhaan committed:
time updated

Nienhuis committed:
added a login screen

user has to type 4 digits and when the digits are correct, the user can
start using the app

mircealungu committed:
set category to watch face

nbhaan committed:
New design

Different wordspace, one big next button below, reveal/remove is now
tapping on the words

nbhaan committed:
Battery + 50/50 layout + rotation background + tap = reveal/remove + next button

Battery level is on the watch, time is white and bigger, background is
rotating from day to night depending on the time, one button on the
screen which is the next button, tap is now for removing/revealing

Nienhuis committed:
Icon for the date

Above the time there is now a icon which print the number of the date

nbhaan committed:
Splitting files + refactoring + more implementation

nbhaan committed:
refactoring + implementing requirejs + adding code

nbhaan committed:
saving/loading user data + flickering canvas + other minor fixes

digeo commented on js/app/time.js:
```diff
@@ -0,0 +1,65 @@
+			minutes = date.getMinutes();
+			seconds = date.getSeconds();
+
+			if (month < 10) {
```
Do not repeat yourself!
create one function:
`function xyz(value) {
   if (value < 10)
       return "0" + value;
return value
}
`

digeo commented on css/style.css:
```diff
@@ -1,37 +1,50 @@
+	width: 170px;
+	height: 120px;
+	border-radius: 20px;
+	background: url('reveal.png');
```
I will write it here.
Move all the images to a new directory (i.e. assets)
They are not css

nbhaan committed:
refactoring + implementing events

nbhaan committed:
r.js as optimizer implemented + events.send

The application is now a lot faster, the login screen is not shown
anymore in case there was already a accountCode stored

Nienhuis committed:
new login screen, menu improved

Login screen is changed to improve usability (the user is now able to
login to their account with a code give on the zeeguu page), the menu
and settings buttons are placed on the bottom of the screen and the user
can now log out in the settings

digeo commented on js/main.js:
```diff
@@ -63,6 +63,9 @@ function printOnScreen(string, posx, posy, size) {
 	ctxWords.fillStyle = "#FFFFFF";
```
create constant: FILL_STYLE_COLOR = "#FFFFFF";

digeo commented on js/main.js:
```diff
@@ -63,6 +63,9 @@ function printOnScreen(string, posx, posy, size) {
 	ctxWords.fillStyle = "#FFFFFF";
 	ctxWords.textAlign = "center";
```
create constant: CTX_WORDS_TEXT_ALIGN = "center";

digeo commented on js/main.js:
```diff
@@ -63,6 +63,9 @@ function printOnScreen(string, posx, posy, size) {
 	ctxWords.fillStyle = "#FFFFFF";
 	ctxWords.textAlign = "center";
 	ctxWords.fillText(string, posx, posy);
+	// print length of word in pixels
```
remove comments of your code

digeo commented on js/main.js:
```diff
@@ -153,6 +153,8 @@ window.onload = function() {
     });
     
     document.getElementById("revealButton").addEventListener("click", function(){
+    	ctxWords.clearRect(0,0, 380,380);
```
extract all the values to constants

digeo commented on .sign/.manifest.tmp:
```diff
@@ -1,137 +1,6 @@
-.DS_Store__DEL__ZZr7nHdvp979KE+Mau6S+zA8+LL3vz/0NlqZKML6klY=
```
please define a good .gitignore!

digeo commented on js/main.js:
```diff
@@ -133,42 +164,50 @@ function printDigitalTime() {
     });
+}
+
+function printFirstWord(){
```
Maybe you should refactor this function to:
`function printWord(word){
    printOnScreen(word SCREEN_MIDDLE, WORD_POSY, MAX_TEXTSIZE);
}`

digeo commented on js/main.js:
```diff
@@ -151,28 +146,27 @@ function printDigitalTime() {
 	ctxTime.fillStyle = BLACK;
 	ctxTime.textAlign = CENTER;
-    ctxTime.fillText(hours + ":" + minutes, SCREEN_MIDDLE, DIGITAL_TIME_POSY);
+    ctxTime.fillText(hours + ":" + minutes, SCREEN_WIDTH/2, DIGITAL_TIME_POSY);
```
create one new constant: `var XYZ = SCREEN_WIDTH/2`

digeo commented on js/main.js:
```diff
@@ -28,17 +28,12 @@ var USERNAME = 'session/i@mir.lu';
 //definitions for coordinates
 var SCREEN_WIDTH = 360;
```
Why do you have hard-coded values for the width of the screen?
Do all the smartwatches have 360 width?

nbhaan committed:
tempVersion with weather

nbhaan committed:
Revert "tempVersion with weather"

This reverts commit 4469826b9d06988a39da14937d261cb56f953ba1.

nbhaan committed:
login, weather, flashcard, saving & sending improved and gui change

nbhaan committed:
improved flashcard + some refactoring + saving & loading bug fixed

nbhaan committed:
improved flashcard

nbhaan committed:
Big Refactoring + clicktracker + fireworks

nbhaan committed:
change icon + deleting assets no longer used

mircealungu committed:
a few refactorings and cleaning some code

mircealungu committed:
rename refactorings

nbhaan committed:
Final version ready for review

Profile, popups, new buttons, added headers, refactoring, small bugs.
Note: adding words functionality is still waiting for implementation on
the server side.

nbhaan committed:
Small bugs

Clear weather space, remove fireworks clicking, medalPos in profile fix.

nbhaan committed:
Clicktracker, assets, context bug, adding words

Sending/saving/loading clicks, removing assets not needed anymore,
context bug, popup too few words, adding words if user learned or
trashed it.

nbhaan committed:
Bug click tracker + removing consoles + profile asset + timeStamps for events

Bug with screenOn event, timeStamp was made before refreshing the time.

nbhaan committed:
removing unnecessary function

nbhaan committed:
fix double events & clicks, getWords optimalisation

nbhaan committed:
adding words async

New words will be fetched on a screenOn asynchronously, they will be
added (if there are any) on a screenOff event.
